build-support/bin/ci.py

```diff
@@ -533,15 +534,44 @@ def run_integration_tests_v2(*, oauth_token_path: Optional[str] = None) -> None:
     default_test_strategy=TestStrategy.v1_no_chroot,
     remote_execution_enabled=oauth_token_path is not None
   )
+
   if target_sets.v2_remote:
-    _run_command(
-      command=TestStrategy.v2_remote.pants_command(
-        targets=target_sets.v2_remote, oauth_token_path=oauth_token_path
-      ),
-      slug="IntegrationTestsV2Remote",
-      start_message="Running integration tests via V2 remote strategy.",
-      die_message="Integration test failure (V2 remote)",
-    )
+    # NB: We add extra logic to work around the limitation that Google RBE tokens expire after 60
+    # minutes. At our current worker count, Our integration test suite frequently takes longer
+    # than 60 minutes to run. So, if the suite fails, we check if this is due to the token expiring
+    # and if so then regenerate a new token and rerun the suite. This strategy works thanks to
+    # remote caching: already completed tests will use their cached results, so we avoid having to
+    # redo too much work (still some inefficiency), and consequently there will be enough workers
+    # to finish the remaining tests.
```

Comment:
As mentioned: I think that we should try to remove this logic as soon as the quota request gets turned around. If you're comfortable landing this in the meantime, then I am too.

Reply:
Oh hm, this approach doesn't work due to our security mechanism prohibiting more than one token per shard attempt. Will wait for the quota increase.